system/mp/mpcomm.cpp

```diff
@@ -2188,6 +2265,10 @@ int CMPConnectThread::run()
                     if (rd == sizeof(connectHdr.id)) // legacy client
                     {
                         legacyClient = true;
+                        connectHdr.hdr.size = sizeof(PacketHeader);
+                        connectHdr.hdr.tag = TAG_SYS_BCAST;
+                        connectHdr.hdr.flags = 0;
+                        connectHdr.hdr.version = MP_PROTOCOL_VERSION;
```

Comment:
ConnectHdr() { }
ConnectHdr connectHdr;
I didn't think so ?

Reply:
I misread - for some reason I though all the ConnectHdr ctor's initialize - clearly ConnectHdr() does not.
If it did initialize cleanly, it would have to initialize 'id' too..

I don't think it's worth changing at this juncture if at all.

I would leave as it was, and as u say, these members are not used in this context, so I'd remove this new initialization code anyway, as it tends to suggest they are imoprtant here.